src/goCommand.ts

```diff
@@ -0,0 +1,43 @@
+import util = require('util');
+import cp = require('child_process');
+import { getBinPath } from './util';
+
+const execFile = util.promisify(cp.execFile);
+
+export class GoCommandResult {
+	stderr: string;
+	stdout: string;
+	err: Error;
+}
+
+export async function goGet(path: string, flags: string[], opts: any): Promise<GoCommandResult> {
+	return goCommand('get', path, flags, opts);
+}
+
+export async function goBuild(path: string, flags: string[], opts: any): Promise<GoCommandResult> {
+	return goCommand('build', path, flags, opts);
+}
+
+async function goCommand(cmd: string, path: string, flags: string[], opts: any): Promise<GoCommandResult> {
+	const goRuntimePath = getBinPath('go');
+	if (!goRuntimePath) {
+		return null;
+	}
+
+	let args: string[] = [cmd];
+	args = args.concat(flags);
+	args.push(path);
```

Comment:
The above can be simplified to
`const args = [cmd, ...flags, path];`

Reply:
Done.